SelectProposal: Add a way to show/hide tabs

mockoocy opened the pull request:
At MAX IV we don't distinguish between different types of sessions (scheduled, unscheduled, other beamline session).
Our sessions are created "on the fly" and are not tied to a particular beam time, thus it's not necessary to split them.

This commit makes it possible to opt-out of displaying some of the tabs via `ui.yaml` config file.

walesch-yan reviewed (approved):
Looks like a good addition to me, thanks @mockoocy

fabcor-maxiv commented on mxcubeweb/core/models/configmodels.py:
```diff
@@ -104,11 +104,31 @@ class UISampleViewVideoControlsModel(UIPropertiesModel):
 
+class UISessionPickerTabModel(BaseModel):
+    id: Literal[
+        "active_tab", "scheduled_tab", "other_beamlines_tab", "all_sessions_tab"
```
Maybe this for consistency, also this would match what is in the documentation:

```suggestion
        "active", "scheduled", "other_beamlines", "all"
```

mockoocy replied:
Done, but I've kept name "all_sessions" instead of just "all" - the latter sounds like it were to just open all the possible tabs

axelboc commented on mxcubeweb/core/models/configmodels.py:
```diff
@@ -104,11 +104,31 @@ class UISampleViewVideoControlsModel(UIPropertiesModel):
+
+class UISessionPickerModel(BaseModel):
+    id: Literal["session_picker"] = "session_picker"
+    components: list[UISessionPickerTabModel] = Field(
```
Could this property be called `tabs` maybe?

axelboc commented on ui/src/components/LoginForm/SelectProposal.jsx:
```diff
@@ -27,6 +38,17 @@ function SelectProposal() {
+  const showTab = Object.fromEntries(tabs.map((tab) => [tab.id, tab.show]));
+
+  const defaultActiveTab =
+    TAB_PRIORITY.find((tabId) => showTab[tabId]) ?? 'scheduled_tab';
```
```suggestion
    TAB_PRIORITY.find((tabId) => showTab[tabId]) ?? TAB_PRIORITY[0];
```

axelboc commented on docs/source/config/ui.rst:
```diff
@@ -273,3 +274,29 @@ Below is an example of a UI configuration file ``ui.yaml``, as used by the mocku
+        - id: other_beamlines
+          show: <show>
+        - id: all
+          show: <show>
```
I feel like this could be simplified to:

```yml
components: # (or tabs:)
  active: <bool>
  scheduled: <bool>
  ...
```

mockoocy replied:
It indeed looks prettier.
I had the `components` section there, because it was required [there](https://github.com/mxcube/mxcubeweb/blob/2eaceacd981fe5b5e9a9a0e387884ad738f3b0e9/mxcubeweb/app.py#L308)

But I've changed this line since.

marcus-oscarsson commented:
Really nice work, Id just like to double check with @antolinos that this is not also overlapping with some of this changes ?

mockoocy committed:
Allow UI layout sections to not have component key

Previously UI section (that is a top-level dictionary in the config) was
required to have a `components` property.
https://github.com/mxcube/mxcubeweb/blob/2eaceacd981fe5b5e9a9a0e387884ad738f3b0e9/mxcubeweb/app.py#L308
It's neither documented nor defined by a Pydantic model.

The `components` section made sense when ui config was referencing some
hardware objects, but it not neceserarily may be the case.

mockoocy committed:
SelectProposal: Add a way to show/hide tabs

At MAX IV we don't distinguish between different types of sessions
(scheduled, unscheduled, other beamline session).
Our sessions are created "on the fly" and are not tied to a particular
beam time, thus it's not necessary to split them.

This commit makes it possible to opt-out of displaying some of the tabs
via `ui.yaml` config file.

antolinos commented:
Dear @mockoocy 

At the ESRF, after several months (almost a year) using the tab system that organizes sessions based on when and where they were scheduled, the feedback we’ve received is perfectly aligned with your PR — we wanted to move away from using tabs and simply display a list of sessions instead.

Basically, we want to have a table with the available sessions and, on top of the table, an easy way to filter the sessions based on different factors such as dates and beamline.

If you're creating sessions on the fly, I don't think this would be useful for you, but I'm wondering if you think it would be compatible with your developments?

mockoocy commented:
> If you're creating sessions on the fly, I don't think this would be useful for you, but I'm wondering if you think it would be compatible with your developments?

I believe it would be.
This PR would allow you to just render the tab containing all sessions. The config would look something like that:
```yaml
# ...
session_picker:
  id: session_picker
  tabs:
    active: false
    scheduled: false
    other_beamlines: false
    all_sessions: true
```
The existing filters are still applied, adding additional ones should still work :)

![image](https://github.com/user-attachments/assets/ffe2ffbf-5083-495d-a3a7-281315a8a282)

marcus-oscarsson commented:
Thanks guys, well done. Looks good !

marcus-oscarsson commented on docs/source/config/ui.rst:
```diff
@@ -273,3 +274,25 @@ Below is an example of a UI configuration file ``ui.yaml``, as used by the mocku
+
+.. _session_picker:
+
+Configures the "Select session" dialog.
```
Nice, thanks for the documentation

marcus-oscarsson commented on ui/src/components/LoginForm/SelectProposal.jsx:
```diff
@@ -11,6 +11,12 @@ import ActionButton from './ActionButton';
 import styles from './SessionTable.module.css';
 
+/**
+ * @typedef {"active"|"scheduled"|"other_beamlines"|"all_sessions"} TabId
```
Nice
